pkg/container-collection/options.go

```diff
@@ -124,8 +132,10 @@ func WithContainerRuntimeEnrichment(runtime *containerutils.RuntimeConfig) Conta
 	return func(cc *ContainerCollection) error {
 		runtimeClient, err := containerutils.NewContainerRuntimeClient(runtime)
 		if err != nil {
-			log.Warnf("Runtime enricher (%s): failed to initialize container runtime: %s",
-				runtime.Name, err)
+			if !cc.disableContainerRuntimeWarnings || log.IsLevelEnabled(log.DebugLevel) {
```

Comment:
What about moving the `IsLevelEnabled` check to ig-manager.go
```
	if !log.IsLevelEnabled(log.DebugLevel) && isDefaultContainerRuntimeConfig(runtimes) {
		warnings := []containercollection.ContainerCollectionOption{containercollection.WithDisableContainerRuntimeWarnings()}
		opts = append(warnings, opts...)
	}
```

Reply:
Done. Looks better like that!